src/sentry/sentry_metrics/use_case_id_registry.py

```diff
@@ -0,0 +1,49 @@
+from __future__ import annotations
+
+from typing import Any, Iterator
+
+_HARDCODED_USE_CASES = {"PERFORMANCE": "performance"}
+
+_REGISTERED_USE_CASES: dict[str, str] = {}
```

Comment:
This is more of a question towards the project in general. How come we are separating `_HARDCODED_USE_CASES` and `_REGISTERED_USE_CASES`? I understand there's a `register_use_case` method, but why can't hard coded use cases be treated as a registered one?

Reply:
agree with John, I think the hardcoded/registered division was kind of a relic in Snuba where old things actually referred to the variables in a bunch of places, and we didn't want to change 100+ files at a time.

here nothing pre-existing is referencing "hardcoded" use cases so you can have them be 100% dynamic